client.go

```diff
@@ -360,4 +482,100 @@ func (c *Client) createComponents(config *Config) {
 		Logger:               c.ic.Logger,
 	}
 	c.proxyManager = newProxyManager(proxyManagerServiceBundle)
+	c.db = sql.OpenDB(idriver.NewConnectorWithClient(c.ic, true))
+}
+
+// SQLOptions are server-side query options.
+type SQLOptions struct {
+	cursorBufferSize int32
+	timeout          *int64
+	schema           *string
+	err              error
+}
+
+/*
+SetCursorBufferSize sets the query cursor buffer size.
+When rows are ready to be consumed, they are put into an internal buffer of the cursor.
+This parameter defines the maximum number of rows in that buffer.
+When the threshold is reached, the backpressure mechanism will slow down the execution, possibly to a complete halt, to prevent out-of-memory.
+The default value is expected to work well for most workloads.
+A bigger buffer size may give you a slight performance boost for queries with large result sets at the cost of increased memory consumption.
+Defaults to 4096.
+The given buffer size must be in the non-negative int32 range.
+*/
+func (s *SQLOptions) SetCursorBufferSize(cbs int) {
+	v, err := check.NonNegativeInt32(cbs)
+	if err != nil {
+		s.err = ihzerrors.NewIllegalArgumentError("setting cursor buffer size", err)
+		return
+	}
```

Comment:
Is there a specific reason not throwing here but while executing SQL? Note: I have read [this thread](https://github.com/hazelcast/hazelcast-go-client/pull/728/files#r786701195) but could not understand the reason of the change in that comment.

Reply:
Throwing a panic is not nice in the public API. We could return an error, but that makes the API harder to use. Also ATM not all setters for SQL options produce errors, but in the future they might.